UNITTESTS/features/netsocket/TCPSocket/test_TCPSocket.cpp

```diff
@@ -29,11 +43,245 @@ class TestTCPSocket : public testing::Test {
 
     virtual void TearDown()
     {
+        stack.return_values.clear();
+        eventFlagsStubNextRetval.clear();
         delete socket;
     }
 };
 
+TEST_F(TestTCPSocket, get_proto)
+{
+    TCPSocketFriend tcpFriend;
+    EXPECT_EQ(tcpFriend.get_proto(), NSAPI_TCP);
+}
+
 TEST_F(TestTCPSocket, constructor)
 {
     EXPECT_TRUE(socket);
 }
+
+TEST_F(TestTCPSocket, constructor_parameters)
+{
+    TCPSocket socketParam = TCPSocket(dynamic_cast<NetworkStack*>(&stack));
```

Comment:
As I'm not that experienced with C++ I would like to know why explicit cast is needed here?

Reply:
To get a runtime exception if casting is not possible.

For example, you modify your stub so that it is not actually inheriting NetworkStack anymore. Dynamic casting catches this error, but static casting does not. It might cause then sefgault.

Reply:
How does this work with RTTI disabled , or UNITTests have RTTI enabled?

Reply:
RTTI is by default enabled in GCC. It can be disabled with -fno-rtti flag, but I do not see it used anywhere in the UNITTESTS directory or in the build files, so I assume it remains enabled.